core/src/main/java/de/evoila/cf/broker/service/impl/BindingServiceImpl.java

```diff
@@ -273,13 +277,58 @@ protected void validateBindingNotExists(String bindingId, String instanceId)
 			} catch (NoSuchElementException e) {
 				return;
 			}
-			if (bindCreation && !isBindingInProgress)
-				throw new ServiceInstanceBindingExistsException(bindingId, instanceId);
-		}
+            if (bindCreation && !isBindingInProgress) {
+                ServiceInstanceBinding serviceInstanceBinding = bindingRepository.findOne(bindingId);
+                boolean identical = wouldCreateIdenticalBinding(serviceInstanceBindingRequest, serviceInstanceBinding);
 
-	}
+                throw new ServiceInstanceBindingExistsException(bindingId, instanceId, identical);
+            }
+        }
+    }
 
 	public ServiceInstance getServiceInstance(String instanceId) throws ServiceInstanceDoesNotExistException {
 		return serviceInstanceRepository.getServiceInstance(instanceId);
 	}
+	private boolean wouldCreateIdenticalBinding(ServiceInstanceBindingRequest request, ServiceInstanceBinding serviceInstanceBinding) throws ServiceInstanceDoesNotExistException {
+		ServiceInstance serviceInstance = serviceInstanceRepository.getServiceInstance(serviceInstanceBinding.getServiceInstanceId());
+		String routeBinding = getRouteBindingFromInstanceBinding(serviceInstanceBinding.getId());
+		String requestRouteBinding = getRouteBindingFromBindingRequest(request);
+		String requestAppGuid = getAppGuidFromBindingRequest(request);
+
+		return Objects.equals(routeBinding, requestRouteBinding) &&
+				Objects.equals(requestAppGuid, serviceInstanceBinding.getAppGuid()) &&
+				Objects.equals(request.getServiceDefinitionId(), serviceInstance.getServiceDefinitionId()) &&
+				Objects.equals(request.getPlanId(), (serviceInstance.getPlanId())) &&
+				Objects.equals(request.getParameters(), serviceInstanceBinding.getParameters());
+	}
+
+	private String getAppGuidFromBindingRequest(ServiceInstanceBindingRequest request) {
+		String appGuid = request.getAppGuid();
+		if (request.getBindResource() != null && request.getBindResource().getAppGuid() != null) {
+			appGuid = request.getBindResource().getAppGuid();
+		}
+		return appGuid;
+	}
+
+	private String getRouteBindingFromBindingRequest(ServiceInstanceBindingRequest request) {
+		BindResource bindResource = request.getBindResource();
+
+		if (bindResource != null) {
+			return bindResource.getRoute();
+		}
+		return null;
+	}
+
+	private String getRouteBindingFromInstanceBinding(String bindingId) {
+		RouteBinding routeBinding;
+		try {
+			routeBinding = routeBindingRepository.findOne(bindingId);
+
+			return routeBinding.getRoute();
+		} catch (NoSuchElementException e) {
+			log.debug("No route for BindingId: " + bindingId + " found", e);
```

Comment:
When creating an identical service instance binding without a route, this passage will always log a NoSuchElementException, which in my opinion lets it appear as there was an error.